src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -101,18 +121,27 @@ public override IEnumerable<StringGetter[]> GetSignatures()
         private partial class UserDefinitionsNameResolver : INameResolver
         {
             private readonly INameResolver _globalNameResolver;
-            private readonly INameResolver _parameterNamedResolver;
+            private readonly Dictionary<string, UDFArg> _args;
```

Comment:
>Dictionary [](http://example.com/codeflow?start=29&length=10)

IReadOnlyDictionary?

Reply:
When do you need to Add to it? Looks like it's just in the constructor. 

Couldn't that be:
```
var argsByName = new Dictionary<string, UDFArg>();
foreach (UDFArg arg in args)
{
    if (!argsByName.ContainsKey(arg.VarIdent.Name.Value))
    {
        argsByName.Add(arg.VarIdent.Name.Value, arg);
    }
}

this._args = argsByName;
```